src/currencyservice/server.js

```diff
@@ -84,60 +84,67 @@ function main () {
    * Lists the supported currencies
    */
   function getSupportedCurrencies (call, callback) {
-    const parentSpan = opentelemetry.getSpan(opentelemetry.context.active());
-    const span = tracer.startSpan('getSupportedCurrencies', { parent : parentSpan });
+    const span = tracer.startSpan("getSupportedCurrencies")
     span.setAttribute('vendor.error_id', '17343337');
-    logger.info('Getting supported currencies...');
-    _getCurrencyData(span, (data) => {
-      callback(null, {currency_codes: Object.keys(data)});
+    try {
+      api.context.with(api.trace.setSpan(api.context.active(), span), () => _getCurrencyData((data) => {
```

Comment:
JW do the emitted spans look any different with these changes or is it basically the same?

Reply:
Good question! I tried to get back to "basically the same" but I didn't examine them in detail. We could do a compare-and-contrast...